docs/guides/js-waku/store-retrieve-messages.md

```diff
@@ -0,0 +1,219 @@
+---
+title: Retrieve Messages Using Store
+---
+
+This guide provides detailed steps to create a light node for retrieving and filtering historical messages using the [Store protocol](/overview/concepts/protocols#store).
+
+## Create a Light Node
+
+Use the `createLightNode()` function to create a light node and interact with the Waku Network:
+
+```js
+import { createLightNode } from "@waku/sdk";
+
+// Create and start a light node
+const node = await createLightNode({ defaultBootstrap: true });
+await node.start();
+```
+
+## Connect to Store Peers
+
+Use the `waitForRemotePeer()` function to wait for the node to connect with store peers:
+
+```js
+import { waitForRemotePeer, Protocols } from "@waku/sdk";
+
+// Wait for a successful peer connection
+await waitForRemotePeer(node, [Protocols.Store]);
+```
+
+## Choose a Content Topic
+
+[Choose a content topic](/overview/concepts/content-topics) for filtering the messages to retrieve and create a `decoder` for [message decryption](https://rfc.vac.dev/spec/26/):
+
+```js
+import { createDecoder } from "@waku/sdk";
+
+// Choose a content topic
+const contentTopic = "/store-guide/1/message/proto";
+
+// Create a message decoder
+const decoder = createDecoder(contentTopic);
+```
+
+## Retrieve Messages
+
+`js-waku` provides the `queryOrderedCallback()` and `queryGenerator()` functions for querying `Store` nodes and retrieving historical or missed messages. The responses from `Store` nodes are paginated and require you to handle them sequentially, processing each page when received.
+
+### `queryOrderedCallback`
+
+The `store.queryOrderedCallback()` function provides a straightforward method for querying `Store` nodes and processing messages in chronological order through a callback function. It accepts these parameters:
+
+- `decoders`: List of `decoders` that specify the `content topic` to query for and their [message decryption](https://rfc.vac.dev/spec/26/) methods.
+- `callback`: The callback function for processing the retrieved messages.
+- `options` (optional): [Query options](/guides/js-waku/store-retrieve-messages#store-query-options) to filter the retrieved messages.
+
+```js
+// Create the callback function
+const callback = (wakuMessage) => {
+	// Render the message/payload in your application
+    console.log(wakuMessage);
+};
+
+// Set the query options
+const queryOptions = {
```

Comment:
In docs.waku.org this an empty object, so maybe let's just not define it?

And only show how to pass it in the "Store Query Options" section?